velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -198,5 +201,33 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, addMonths) {
+  const auto parseDate =
+      [](const std::optional<const char*> date) -> std::optional<int32_t> {
+    if (date.has_value()) {
+      return DATE()->toDays(date.value());
+    }
+    return std::nullopt;
+  };
+
+  const auto addMonths =
+      [&](const std::optional<const char*> date,
+          std::optional<int32_t> value) -> std::optional<std::string> {
+    auto res = evaluateOnce<int32_t, int32_t>(
+        "add_months(c0, c1)", {parseDate(date), value}, {DATE(), INTEGER()});
+    if (res.has_value()) {
+      return DATE()->toString(res.value());
+    }
+    return std::nullopt;
+  };
+
+  EXPECT_EQ("2015-02-28", addMonths("2015-01-30", 1));
+  EXPECT_EQ("2015-01-30", addMonths("2015-01-30", 0));
+  EXPECT_EQ("2016-02-29", addMonths("2016-03-30", -1));
+  EXPECT_EQ(std::nullopt, addMonths("2015-01-30", std::nullopt));
+  EXPECT_EQ(std::nullopt, addMonths(std::nullopt, 1));
```

Comment:
Remove null test cases. They test the framework, not the function. Please, add test cases for overflow.